Aptoide/aptoide-client-v8
# [task/MOB-918] List Apps Updates Improvements

**What does this PR do?**

   This PR attempts to improve the function of list apps updates by avoiding calling the package manager unnecessarily. It also has some minor refactors (such as centralizing room migration).
   I've wasted a lot of time trying to implement other things (room testing, and a redundant call on listAppsUpdates sync on first launch). But both of these things would be too time consuming / non-linear to fix, and I've spent too much time on this ticket already.
  From benchmarking what is implemented here, it didn't make much difference other than having "more correct" code.
 
**Database changed?**

  No

**How should this be manually tested?**

  Check if listAppsUpdates is functioning still as expected (do a first launch). Also, at first launch, it should NOT duplicate listAppsUpdates requests now (before it would duplicate the requests).

**What are the relevant tickets?**

  [MOB-918](https://aptoide.atlassian.net/browse/MOB-918)


**Code Review Checklist**
